### PR TITLE
Favorites to Question

### LINKED ISSUE #25: Add favorite attribute to question
Ensure that local questions can be marked as a favorite. This state will never be shared with the remote server.

### PR DESCRIPTION
**Pull request by linster**
Resolved #25 

The addFavoriteQuestions takes a UUID. When doing answer favorites, FOR THE LOVE OF GOD refactor the datamanager and don't copy and paste code. UUIDs work just fine.

### CODE REVIEW

**Comment by sdwowk** [comment]
pretty pictures
Looks good but get rid of comment about shared preference magic!

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/ClientData.java [hunk 1]
```diff
@@ -48,11 +51,50 @@ public void clear() {
...
-		ArrayList<UUID> list = new ArrayList<UUID>();
-		return list;
+		//Supposed to return a list of favorite questions
+		//We can then add to this list
+		
+	
+		
+		//Pull a list of favorited questions from SharedPrefs
+		
+		//TODO: SharedPreferences Magic!!!
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/ClientData.java]: You could clean these up.

**Comment by linster** [comment]
See commit 2.

**Comment by stephenjust** [comment]
Looks fine.

**Pull request merged**